torch/_dynamo/variables/user_defined.py

```diff
@@ -250,9 +250,12 @@ def call_method(
             if method is object.__init__:
                 return ConstantVariable.create(None)
 
-            if method is collections.OrderedDict.keys and self.source:
+            # [NOTE] OrderedDict, dict subtypes must always have source
+            # We cannot instantiate such subtypes in-graph due to builtin __new__
```

Comment:
I mean, in principle we could if we implemented their `__new__` correctly, no?

Reply:
yes, but I didn't want to deal with that case if it is not on usage path.

Reply:
Instantiating __new__ directly is very rare so we assume it never happens.